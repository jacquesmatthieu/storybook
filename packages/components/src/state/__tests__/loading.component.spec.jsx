import React from 'react';
import { shallow } from 'enzyme';

import Loading from '../loading.component';

jest.mock('../withCurrentApp.hoc', () => jest.fn(Comp => Comp));

describe('Loading', () => {
  let props;

  beforeEach(() => {
    props = {
      currentApp: { id: 'foo' },
    };
  });

  const getLoadingWrapper = () => shallow(<Loading {...props} />);

  it('should render properly', () => {
    // When
    const wrapper = getLoadingWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error', () => {
    // Given
    props.error = 'foo';

    // When
    const wrapper = getLoadingWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
