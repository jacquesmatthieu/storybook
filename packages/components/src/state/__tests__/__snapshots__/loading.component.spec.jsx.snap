// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render error 1`] = `
<AppError
  error="foo"
  t={[MockFunction]}
/>
`;

exports[`Loading should render properly 1`] = `
<AppLoading
  appId="foo"
/>
`;
